### DISCUSSION TITLE
"npx create-remix@latest" doesn't work

### DISCUSSION THREAD

**Discussion by pagebanc** [remix-run/remix#2709] [General]
When I ran `npx create-remix@latest` with the following options, the project wasn't created. 
It worked fine the other day but not working today (4/8/2022).

The options that I selected:
```
? Where would you like to create your app? ./my-remix-app
? What type of app do you want to create? Just the basics
? Where do you want to deploy? Choose Remix if you're unsure; it's easy to change deployment targets. Fly.io
? TypeScript or JavaScript? TypeScript
? Do you want me to run `npm install`? No
```

Then, I got the following error:
```
⠹ Creating your app…🚨 The provided template must be a Remix project with a `package.json` file, but that file does not exist in...
```

Another quick question.
Will `npx create-remix@latest` create the application structure using **React 18** soon? 
When?

Thank you.

**Comment by tomvorel13** [comment]
Same issue...

**Comment by ghost** [comment]
yes, Same ! use remix v1.3.5 + express

**Comment by MrWittman612** [comment]
same here I'm on windows if that makes a difference. thought I was going crazy!!!!!!!

**Reply by MrWittman612** [reply]
I was able to get a project up by running npx create-remix --template remix-run/indie-stack "my-app" lot of boiler plate though.

**Comment by f-bog** [comment]
I also thought I was going crazy. On a windows machine as well

**Comment by qianjiachun** [comment]
Try the following:  
1. `npx create-remix@1.3.4 remix-demo`
2. options:
```
? What type of app do you want to create? Just the basics
? Where do you want to deploy? Choose Remix if you're unsure, it's easy to change deployment targets. Remix App Server
? TypeScript or JavaScript? TypeScript
? Do you want me to run `npm install`? Yes
```
3. `npm-check -u`
4. choose remix 1.3.5 and update (React18 can also be upgraded)
  
On my computer (Windows), vscode will report many errors after upgrading dependency. I wonder if you have encountered such a problem😂  
That's how I solved it:  
1. `rimraf .\node_modules`
2. `npm install`

**Comment by anantoghosh** [comment]
@kentcdodds the recent changes to `create.ts` has caused this issue. febf2bf391b8cd15dfaa06c83fbd33cf0e814fdf
https://github.com/remix-run/remix/blob/7d799e2b6d71f5d7ff572addca623545f975af96/packages/remix-dev/cli/create.ts#L255-L271

At least on windows, `header.name.startsWith(filePath)` is always false causing all files to get ignored.

**Reply by nunoleong** [reply]
crappy workaround (with WSL): open a `bash` and it'll work

**Reply by ArifRabbaniIn** [reply]
Same Issue

**Reply by obillwan** [reply]
I got the same message.  I'm using Windows 10.

**Reply by Dan86de** [reply]
Same issue here

**Reply by hasanmn** [reply]
same issue here, on Windows 10

**Comment by machour** [comment]
Tracked in https://github.com/remix-run/remix/issues/2714

**Comment by yjavaherian** [comment]
same here using windows 11 build 22000.469

**Comment by appsdevpk** [comment]
same issue on windows 10, it also didn't asked the choice of javascript or typescript

**Comment by ArifRabbaniIn** [comment]
Tried with v1.4.0. same issue

**Reply by machour** [reply]
@arabbani the fix will lend in 1.4.1

**Reply by ArifRabbaniIn** [reply]
Great.

**Comment by HosseinZafari** [comment]
Same issue on Windows 11 (With CMD And PowerShell)

**Comment by machour** [comment]
Can y'all try with this pre-release?

```
npx create-remix@1.4.1-pre.1
```

**Reply by ElvisKnapman** [reply]
For me at least, I just tried using `npx create-remix@1.4.1-pre.1` and it seems to fix the problems I had running `npx create-remix@latest`

**Reply by HosseinZafari** [reply]
I tried it and we have another issue now

```shell
npx create-remix@1.4.1-pre.1 test
? What type of app do you want to create? Just the basics
? Where do you want to deploy? Choose Remix if you're unsure; it's easy to change deployment targets. Remix App Server
? Do you want me to run `npm install`? Yes
\ Creating your app…filePath.replaceAll is not a function
```

**Reply by machour** [reply]
@HosseinZafari see the reply for your case here: https://github.com/remix-run/remix/issues/2714#issuecomment-1102865708

**Reply by iwan-uschka** [reply]
Same issue here. @machour if you mean
> For now you can use npx create-remix@1.3.4

then yes, that works.

**Comment by machour** [comment]
1.4.1 is out, and fixed some issues on windows. 

If you're still facing problems with that version, please report it to #2714 instead of here, it will help the team tremendously 🙏

**Comment by Loschcode** [comment]
```
❯ nvm use 18
Now using node v18.16.1 (npm v9.5.1)
❯ npx create-remix@latest

 remix   v2.6.0 💿 Let's build a better website...

   dir   Where should we create your new project?
         /website

      ◼  Using basic template See https://remix.run/guides/templates for more
      ✔  Template copied
```

Not working on my end, there's no `website` anywhere despite the CLI telling me otherwise

**Reply by kiliman** [reply]
Don't add the `/` at the start of your project path. It's probably trying to create a folder in the root of your file system, which you typically don't have permission to 

Just enter `website` and it will create the project folder where you ran the command.
